src/main/java/com/openlattice/controllers/OrganizationsController.java

```diff
@@ -212,14 +208,22 @@ public OrganizationIntegrationAccount getOrganizationIntegrationAccount( @PathVa
     public Map<UUID, Set<OrganizationEntitySetFlag>> assembleEntitySets(
             @PathVariable( ID ) UUID organizationId,
             @RequestBody Set<UUID> entitySetIds ) {
+        // materialize should be a property level permission that can only be granted to organization principals and
+        // the person requesting materialize should be the owner of the organization
         ensureOwner( organizationId );
-        final var authorizedPropertyTypesByEntitySet =
-                authzHelper.getAuthorizedPropertiesOnEntitySets( entitySetIds, EnumSet.of( Permission.MATERIALIZE ) );
+
         final var organizationPrincipal = organizations.getOrganizationPrincipal( organizationId );
         if ( organizationPrincipal == null ) {
             //This will be rare, since it is unlikely you have access to an organization that does not exist.
             throw new ResourceNotFoundException( "Organization does not exist." );
         }
+
+        entitySetIds.forEach( entitySetId -> ensureMaterialize(entitySetId, organizationPrincipal) );
+        final var authorizedPropertyTypesByEntitySet = authzHelper.getAuthorizedPropertiesOnEntitySets(
```

Comment:
I don't think we want to check materialize on the the property types -- as long as the person has materialize permissions on the entity set aclKey, the full entity set should be materialized, and then the correct permissions will be enforced on reading the materialized entity set. Since the organization shares a single materialized view for an entity set, partially materializing will lead to problems for people who do have full permissions.

Reply:
I followed the documentation on this, but maybe I misunderstood: https://wiki.openlattice.com/pages/viewpage.action?spaceKey=ENGR&title=Materialized+Entity+Set+Views
As I understand: the owner of org can add/revoke materialize permissions to/from the org principal

Reply:
oh you're right -- this may currently be problematic since I don't think there's any way to grant permissions to an organization right now (other than postman) so nothing will be materializable until we make those updates. Also we are currently granting / making people able to grant materialize to users and roles, and removing that may introduce some new problems in cases where we're checking against EnumSet.allOf( Permission.class) , etc